metal/resource_metal_device.go

```diff
@@ -46,9 +47,22 @@ func resourceMetalDevice() *schema.Resource {
 			},
 
 			"hostname": {
-				Type:        schema.TypeString,
-				Description: "The device name",
-				Required:    true,
+				Type:          schema.TypeString,
+				Description:   "The device name",
+				Optional:      true,
+				ForceNew:      true,
```

Comment:
I'm reluctant to accept the ForceNews. Why do suppose this is needed?

@t0mk any thoughts?

Reply:
Overall, I believe it would be better for the provider if that field was immutable and replace the resource when it gets changed. In terms of backward compatibility, even if it changes the API, I don't believe that it will have much of an impact on users.

It removes a bit of logic on Update to rename the device on hostname change. In practice, I haven't seen devices being renamed to I suspect that this part of the code is buggy already.

If the device hostname changes, it won't reflect on the OS hostname automatically. I would rather see the device getting replaced and both map 1:1 all of the time.